Atomic.md

```diff
@@ -0,0 +1,270 @@
+Atomic placeholder operations in VFSForGit on MacOS
+===================================================
+
+A potential hole in our current design is in our creation of
+placeholders. Today, when creating a placeholder directory, at a high
+level, we go through the following workflow:
+
+-   Create placeholder folder/file in the virtualization root
+
+-   'Mark up' the file with the proper flags/xattrs
+
+-   (If creating a file) Update the mode of the file
+
+This could result in scenarios where a partial placeholder appears in
```

Comment:
What do you mean by "partial placeholder"?  If you mean a placeholder that is missing some/all of its metadata, consider using a term like "incomplete placeholder" to avoid confusion with the "partial files" term on Windows.